openrtb_ext/request_wrapper.go

```diff
@@ -239,6 +239,7 @@ func (rw *RequestWrapper) rebuildImp() error {
 			return err
 		}
 		rw.Imp[i] = *rw.impWrappers[i].Imp
+		rw.impWrappers[i].Imp = &rw.Imp[i]
```

Comment:
@hhhjort we discussed removing `rw.Imp = make([]openrtb2.Imp, len(rw.impWrappers))` and `rw.Imp[i] = *rw.impWrappers[i].Imp` but I want to talk to Scott about this because I saw that the unit test was written to handle the scenario where `req.Imp` does not match `req.impWrappers` imps which we said should theoretically never happen unless something is broken. We can change to that implementation in a future commit once I confirm with Scott that we aren't missing something.

Reply:
Yes, we will need to modify `SetImps()` to rebuild the underlying Imp[] array in the base request so that the imp wrappers stay in sync with the base imps. Something similar to the lines we removed from rebuidImps, with an added step to reset the pointers in the imp wrappers to the imps in the new array we just created.